app/controllers/pupils/schools_controller.rb

```diff
@@ -1,9 +1,42 @@
 class Pupils::SchoolsController < SchoolsController
   def show
+    @activities_count = @school.activities.count
+    @first = @school.activities.empty?
+    @completed_activity_count = @school.activities.count
+    @suggestion = NextActivitySuggesterWithFilter.new(@school, activity_type_filter).suggest.first
+
     @latest_alerts_sample = @school.alerts.usable.latest.sample(2)
     @scoreboard = @school.scoreboard
     if @scoreboard
       @surrounding_schools = @scoreboard.surrounding_schools(@school)
     end
+    # Temporary until equivalences are set up
+
+    if @school.meters?(:electricity)
+      electricity_card = MeterCard.create(school: @school, supply: :electricity)
```

Comment:
Rather than creating the card, just call the class method:   def calulate_average_usage(school, supply, window)
And whilst we're at it, change them to be named params